### PR TITLE
Add bubble plots to Python

### LINKED ISSUE #43: Add bubble map support for plotting
Currently only choropleths are supported with plotting. Add support for bubble maps like what is available in the [R package](https://cmu-delphi.github.io/covidcast/covidcastR/articles/plotting-signals.html#bubble-maps)

### PR DESCRIPTION
**Pull request by chinandrew**
Closes #43 

Summary of changes:
- Add bubble map plotting option.
- Update interface from `plot_choropleth()` to `plot()`, with the `plot_choropleth()` still available with a deprecationwarning. `plot()` defaults to `plot_type="choropleth"`. 
- 

On bubble sizing, it's set so the maximum bubble size is proportional to the area of the figure, with 8 bubbles scaled by area between 0.1 and the mean + 3std dev.

On interface: I failed to realize that a bubble map was distinct from a choropleth, so having `plot_choropleth(plot_type='bubble')` isn't technically accurate. A few options:

1. Keep as is.
2. Add it as `plot_bubble()`
3. Change `plot_choropleth()` to `plot()` with the `plot_type` argument. Could keep `plot_choropleth()` as a deprecated function.

Currently have gone with 3, though I don't feel particularly strongly about any of them. I like 3 if we were starting from scratch, but if it's too annoying to update the interface/deprecate it, 2 would be my next choice. If people won't nitpick I also don't see a huge problem with 1. 

TODO
- [x] add tests
- [x] update documentation in docstrings and sphinx
- [x] debug issues where state plots throw error and don't show a legend (happens if you have a df with all nans)
- [x] figure out how to set min/max bubble size based on figsize

state
`covidcast.plot(df,  plot_type = "bubble")`     
Screenshot from 2020-10-01 16-39-30

`covidcast.plot(df, figsize = (6,4), plot_type = "bubble", color ="red") `
Screenshot from 2020-10-01 16-41-48

county  
`covidcast.plot(df,  plot_type = "bubble")`
Screenshot from 2020-10-01 16-42-25

### CODE REVIEW

**Comment by chinandrew** [comment]
Here's something annoying about bubble plots. Since I have them plot the bubble at the centroid of the polygon, if you have a polygon which is contained in another one, like HRR, the bubble for the outer polygon isn't in a useable location:
Screenshot from 2020-10-01 01-29-44

**Comment by krivard** [comment]
@chinandrew is this PR still needed? if so, resolve conflicts and I'll take a look.

**Comment by chinandrew** [comment]
> @chinandrew is this PR still needed? if so, resolve conflicts and I'll take a look.

Thanks, conflicts resolved.

File: Python-packages/covidcast-py/covidcast/plotting.py [hunk 1]
```diff
@@ -64,45 +65,52 @@ def plot_choropleth(data: pd.DataFrame,
     documentation
     <https://geopandas.org/reference.html#geopandas.GeoDataFrame.plot>`_.
 
+    Bubble maps use a purple bubble by default, with all values discretized into 8 bins between 0.1
+    and the signal's historical mean value + 3 standard deviations. Values below 0 are have no
```
Comment (krivard) [inline Python-packages/covidcast-py/covidcast/plotting.py:69]:
```suggestion
    and the signal's historical mean value + 3 standard deviations. Values below 0 have no
```

File: Python-packages/covidcast-py/covidcast/plotting.py [hunk 2]
```diff
@@ -201,14 +209,88 @@ def animate(data: pd.DataFrame, filepath: str, fps: int = 3, dpi: int = 150, **k
...
+    # this is to remove the set_array error that occurs on some platforms
+    sm._A = []  # pylint: disable=W0212
+    plt.colorbar(sm, ticks=np.linspace(kwargs["vmin"], kwargs["vmax"], 8), ax=ax,
+                 orientation="horizontal", fraction=0.02, pad=0.05)
+
+
+def _plot_bubble(ax: axes.Axes, data: gpd.GeoDataFrame, geo_type: str, **kwargs: Any) -> None:
+    """Generate a bubble map on a given Figure/Axes from a GeoDataFrame.
+
+    The maximum bubble size is set to the figure area / 1.5, with a x3 multiple if the geo_type
```
Comment (krivard) [inline Python-packages/covidcast-py/covidcast/plotting.py:245]:
```suggestion
    The maximum bubble size is set to the figure area / 1.5, with a x3 multiplier if the geo_type
```

**Review by krivard — APPROVED** [review]
Minor docs tweaks but otherwise good!

**Review by krivard — APPROVED** [review]
japprove!

**Pull request merged**